src/client.rs

```diff
@@ -36,22 +37,74 @@ pub struct Client {
     inner: Arc<Inner>,
 }
 
+#[derive(Default)]
+pub struct ClientBuilder {
```

Comment:
Can we use [derive_builder](https://crates.io/crates/derive_builder) to reduce boilerplate code?

Reply:
[derive_builder](https://crates.io/crates/derive_builder) will return Result. but we don't need it.